docs/README.md

```diff
@@ -62,6 +62,52 @@ Alternatively, you can use the `scripts/build_linux.sh` script to perform the sa
 ./scripts/build_linux.sh
 ```
 
+### Code Formatting and Pre-commit Hooks
+
+This project uses [pre-commit](https://pre-commit.com) to ensure code quality and consistent formatting.
+
+To set it up:
+
+1. Install `pre-commit` if you don't have it:
+
+   ```bash
+   pip install pre-commit
+   ```
+
+2. Install clang-format:
+
+   ```bash
+   apt install clang-format
```

Comment:
what about Windows developers ?

Reply:
I mean this instruction its for Linux developers, so its OK, but please mention that clearly. 
Now think that some Users would like to work on MacOSX or Windows, just to make some edits in the documentation or update the tests. What then ?